src/NuGet.Clients/PackageManagement.VisualStudio/Migrate/ProjectJsonToPackageRefMigrator.cs

```diff
@@ -11,35 +11,50 @@
 using System.Linq;
 using System.Threading.Tasks;
 using Task = System.Threading.Tasks.Task;
+using Microsoft.VisualStudio.Threading;
 using NuGet.PackageManagement.UI;
+using Microsoft.VisualStudio.Shell;
 
 namespace NuGet.PackageManagement.VisualStudio
 {
     public static class ProjectJsonToPackageRefMigrator
     {
         public static async Task MigrateAsync(LegacyCSProjPackageReferenceProject project, string dteProjectFullName)
         {
+            await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
```

Comment:
@AArnott In my personal experience, I've seen lot of performance and responsiveness issues with being on UI thread default and specially with async client applications. Which is why when I surely know i dont need UI thread, I better be with BG thread. But I also get our point, it could be more prone to hangs and crashes which is why we usually try to evaluate and comes up with performance numbers with UI vs BG thread if there isn't much difference then be with UI thread as default.

About this case, I'm fine with switching to UI thread, but advised to move it inside JTF.RunAsync.